### ISSUE TITLE
WARNING: No actions completed, waiting on build_web_compilers|ddc

### ISSUE THREAD

**Issue by chalin** [dart-lang/build#1375]
Attempting to build most of the example apps results in the following (e.g., see this [Travis log](https://travis-ci.org/dart-lang/site-webdev/jobs/371725070)):

```nocode
architecture > pub run build_runner build --fail-on-severe --delete-conflicting-outputs --output=build
[INFO] Generating build script completed, took 309ms
[INFO] Building new asset graph completed, took 677ms
[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms
[WARNING] No actions completed for 15.1s, waiting on:
  - build_web_compilers|ddc on package:analyzer/analyzer.module
  - build_web_compilers|entrypoint on web/main.dart

[INFO] Running build completed, took 41.9s
[INFO] Caching finalized dependency graph completed, took 118ms
[INFO] Creating merged output dir `build` completed, took 1.5s
[INFO] Writing asset manifest completed, took 1ms
[INFO] Succeeded after 43.7s with 1289 outputs (1724 actions)
chalin@CC5K16 Thu Apr 26 16:26:14 ~/git/wd/examples/ng/doc/architecture (chalin-2-dev-50-sdk-0425)
```

Environment:
- **Dart VM version: 2.0.0-dev.50.0** on "macos_x64"

It was working fine under Dart 2-dev.49.

cc @kwalrath

**Comment by natebosch** [comment]
I think this is more likely to be caused by a change in `build_modules` thank in the SDK. @matanlurey was also seeing some problems with trying to run DDC on the analyzer package which I think we don't normally expect...

I wonder if the course modules are triggering some condition in angular which is grouping compiler code in with runtime code or something.

cc @grouma

**Comment by grouma** [comment]
What exactly are you trying to build? I have tried to build a handful of examples and I can't reproduce the problem with Dart VM version: 2.0.0-dev.50.0 on "macos_x64"

**Comment by grouma** [comment]
Scratch that. I was able to reproduce the problem building hacker_news_pwa on
Dart VM version: 2.0.0-dev.50.0 on "macos_x64"

but not:
Dart VM version: 2.0.0-dev.50.0 (Unknown timestamp) on "linux_x64"

**Comment by chalin** [comment]
@grouma - ok well, in case you'd like to have other examples: try the apps from the `chalin-build-runner-upgrade-0426` branch, such as https://github.com/dart-lang/site-webdev/tree/chalin-build-runner-upgrade-0426/examples/ng/doc/architecture.

**Comment by natebosch** [comment]
We tracked down the issue.

The code to compute modules was ported from pub. During the transition we switched some logic which created a merged module for files in `src` based on a name combining all of the 'entrypoint's that depend on it, and then later naming it back to something sensible.

In the new code we erroneously use a module with the ID of an arbitrary entrypoint that depends on it.
The reason we see different behavior between mac and linux is that the order of the entrypoints is different so `entrypoints.first` gives a different result and we're merging the sources in to one module or another.

https://github.com/dart-lang/build/blob/b50271c08608cb4b0e8a69684d99ddea858cc789/build_modules/lib/src/meta_module.dart#L213

vs

https://github.com/dart-lang/pub/blob/16cce79a2454d3425ebcce42479e93c5b314acce/lib/src/dartdevc/module_computer.dart#L340
